message-signing-spec.md

```diff
@@ -0,0 +1,206 @@
+
+The `<state>` value must be a [UTF-8](https://developer.mozilla.org/en-US/docs/Glossary/UTF-8) string value, and must contain a [MAC](https://en.wikipedia.org/wiki/Message_authentication_code). The wallet will pass this value back to the server in the [PUT request](https://github.com/bedrock-foundation/solana-pay/edit/master/SPEC.md#put-request) in order to verify that the contents of the `<data>` field were not modified.
+
+
+The application may also include an optional `message` field in the response body:
+```json
+{"message":"<message>","data":"<data>","state":"<state>"}
+```
+
+The `<message>` value must be a UTF-8 encoded string that describes the nature of the sign-message response.
+
+For example, this might be the name of the application with which the user is interacting or context about how the sign-message request is being used. The wallet must display at least the first 80 characters of the message field to the user if it is included in the response.
+
+The wallet and application should allow additional fields in the request body and response body, which may be added by future specification.
+
+#### PUT Request
+
+The PUT request is used to send the results of signing the message back to the server. The wallet must make an HTTP `PUT` JSON request to the URL with a body of
+```json
+{"account":"<account>","data":"<data>","state":"<state>","signature":"<signature>"}
+```
+
+The `<account>` value must be the base58-encoded public key of the account that signed the message.
+
+The `<data>` value must be the unmodified `<data>` value from the response of the preceding POST request.
+
+The `<state>` value must be the unmodified `<state>` value from the response of the preceding POST request.
+
+The `<signature>` value is the base58-encoded signature from signing the `<data>` field with the users private key.
+
+The wallet should make the request with an [Accept-Encoding header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding), and the application should respond with a [Content-Encoding header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) for HTTP compression.
+
+The wallet should display the domain of the URL as the request is being made. If a `GET` request was made, the wallet should also display the label and render the icon image from the response.
+
+#### PUT Response
+
+The wallet must handle HTTP [client](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) and [server](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses) errors in accordance with the [error handling](#error-handling) specification. [Redirect responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages) must be handled appropriately. The application must respond with these, or with an HTTP `OK` response. An HTTP `OK` response indicates that signature verification was successful.
```

Comment:
I considered adding an optional `message` field in the PUT response that would be displayed to the user upon successful verification (wallet will know success based on HTTP status code). This would allow dapps to send a thank you note or add additional context to the sign-message experience. It also provides consistency to the dev experience as all valid responses would allow for a `message` field. 

I ultimately left it out as I thought it may impact UX. Thoughts?